lib/rules/no-extra-boolean-cast.js

```diff
@@ -63,6 +63,53 @@ module.exports = function(context) {
                     grandparent.callee.name === "Boolean")) {
                 context.report(node, "Redundant double negation in Boolean constructor call.");
             }
+        },
+        "CallExpression": function(node) {
+            var parent = node.parent;
+
+            if (node.callee.type !== "Identifier" || node.callee.name !== "Boolean") {
+                return;
+            }
+
+            // if (<bool>) ...
+            if (parent.type === "IfStatement") {
```

Comment:
Can you collapse these conditions? You're doing the same thing every time, it would be better to just to do it once.